src/cuda/compute_stacey_viscoelastic_cuda.cu

```diff
@@ -84,13 +84,9 @@ __global__ void compute_stacey_elastic_kernel(realw* veloc,
 
       i = abs_boundary_ij[INDEX3(NDIM,NGLLX,0,igll,iface)]-1;
       j = abs_boundary_ij[INDEX3(NDIM,NGLLX,1,igll,iface)]-1;
-      //check if the point must be computed
-      if (i==NGLLX || j==NGLLX) return;
```

Comment:
Hi Daniel,

I don't think this one is an error, in the 2D CPU version, the array abs_boundary_ij is created such that duplicated points have either i= NGLLX or j = NGLLZ, so the stacey condition is not applied twice to the same gll point. If this happens, we risk incorrect results, or even a seg fault.

Reply:
good point, thanks for checking! 

let me fix it for now. i realized that in the 2D version the boundary conditions are having this somewhat more complex behavior, with the idea of being applied only once for a gll point. that is, it's applying the ABC on a global level. this could be made much simpler by having the condition applied on a local level, avoiding the need for all these special cases for corners and shared gll points. i'll test the performance of the local version versus this global implementation and submit a new PR soon.

Reply:
I agree, the 2D code could and should be much simpler. I also don't like the multiple 4 different arrays for each side, they could be merged. Then, the "cote_abs" array and unnecessary multiple if test could disappear.

Reply:
ok, i'll work on it. for now, the indexing has been put back in PR #1052